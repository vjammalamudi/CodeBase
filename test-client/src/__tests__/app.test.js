import React from 'react';
import { render } from "enzyme";
import App from '../app';

it('renders correctly', () => {
  const rendered = render(<App/>)
  expect(rendered).toMatchSnapshot();
})
