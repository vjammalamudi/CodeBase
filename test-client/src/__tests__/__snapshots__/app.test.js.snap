// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    class="topnav"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Task One
    </a>
    <a
      href="/task-two"
    >
      Task Two
    </a>
  </nav>,
  <div
    class="task"
  >
    <h1>
      Task One
    </h1>
    <div
      class="buttons"
    >
      <button
        class=""
        type="button"
      >
        <img
          alt="whale"
          src="whale.jpg"
        />
        <span>
          Whale
        </span>
      </button>
      <button
        class=""
        type="button"
      >
        <img
          alt="cow"
          src="cow.jpg"
        />
        <span>
          Cow
        </span>
      </button>
      <button
        class=""
        type="button"
      >
        <img
          alt="hippo"
          src="hippo.jpg"
        />
        <span>
          Hippo
        </span>
      </button>
    </div>
    <div
      class="buttons"
    >
      <button
        type="button"
      >
        <span>
           Log my click to console
        </span>
      </button>
    </div>
    <div
      class="content"
    >
      <h4>
        Complete the following task:
      </h4>
      <p>
        This task is to solve a bug. There is a bug in the LogSelectionButton React Class Component that is preventing the behaviour described below from working correctly. Fix the bug.
      </p>
      Expected:
      <ol
        type="1"
      >
        <li>
          LogSelectionButton must remain a Class Component.
        </li>
        <li>
          When you click on an animal tile (Whale, Cow or Hippo) e.g. Hippo then click the 
            "Log my click to console" button then a corresponding message e.g. “You clicked: Hippo” 
            is logged to the Javascript console after 5 seconds.
        </li>
        <li>
          When you click all three animal tiles, within 5 seconds, with a click of the 
          "Log my click to console" button between each animal tile click, a corresponding message 
          is logged to the Javascript console for each animal tile.
        </li>
      </ol>
    </div>
  </div>,
]
`;
